toc/rfc/rfc-draft-add-cf-supports-for-fips-stemcell.md

```diff
@@ -0,0 +1,58 @@
+
+This RFC only captures the work required to get cf-deployment in a state where it can run on a stemcell build with FIPS packages. All components actually using those FIPS packages are out of scope for this RFC.
+
+## Workstreams
+The major milestones have been documented below in more details per working group.
+
+### Foundational Infrastructure WG
+
+
+#### CFF internal FIPSstemcell
+As a first step the `bosh-linux-stemcell-buider` will need to be extended with support for building a Jammy FIPS stemcell variant. As an initial target, support for AWS, GCP and Azure should be added. Once the foundation internal FIPS stemcell can be built the next step should be an automation which can produce stemcells for CFF internal validation by using the FIPS packages provided by Canonical. The stemcells can be made available to other WGs via a private bucket. It is important to notice that this stemcell must not be distributed for public consumption.
+
+#### BOSH supports FIPS stemcell
+As the next step the Foundational Infrastructure WG should validate that the BOSH director can run on a FIPS based stemcell. This can be done by using the BOSH director acceptance test. The validation should be automated and used for every candidate release of the Director.
+
+
+#### Reproducible build for stemcells (Optional)
+While compatibility testing is underway, a second track of work can be picked up, which has to do with the fact that the `bosh-linux-stemcell-buider` currently does not produce reproducible artifacts. The various Ubuntu packages that are installed during the build process are not pinned down, which is great from a security perspective, but not great if downstream vendors want to build a FIPS stemcell version internally using their own Ubuntu Pro license.
+
+
+Without being too prescriptive, the desired outcome would be some sort of snapshot file containing the package versions being committed to the stemcell-builder repository, before actually building a stemcell. Effectively decoupling building from the act of bumping package versions. Package bumps should be automated and triggered using the same USN feed used to trigger builds now. This functionality should also be contributed to the non-FIPS Jammy pipeline.
+
+### App Runtime Deployments WG
+
+Once initial confidence has been gained by validating the FIPS stemcell against the BOSH director it is time to validate against cf-deployment. Just like with the BOSH director, an automation should be set up to run the CATS test suite against a Cloud Foundry environment which uses the FIPS stemcell from the private foundation bucket. Issues discovered by this validation should be distributed to the related CF components. It is their responsibility to address them. If CF components want to have faster feedback on FIPS issues they can add their own validation.
```

Comment:
We could add here that we will run the validation on the AWS infrastructure. If needed, we can also validate on GCP. Azure is currently out of scope.

Reply:
@jochenehret I added some sentences about the point you raised. Please check if it is OK like this for the cf-deployment WG.